### PR TITLE
Round difference between desiredAmount and currentAmount to 2 decimals

### PR DESCRIPTION
**Pull request by domangi**
round difference between desiredAmount and currentAmount to manage javascript number precision and have the expected exact difference

### CODE REVIEW

**Comment by roschaefer** [comment]
At the moment, broadcasting fees in Germany are at 17.50€/month per home. The backend will not accept a user who is distributing more than the maximum budget.

Right now, a user can specify the desired amount for every supported broadcast. The frontend will check:
1. Is the desired amount <= 17.50 ?
2. Is the sum of all amounts <= 17.50?
3. If 2) does not hold, can we reduce the amounts of the other broadcasts to make 2) pass?

If this is not possible, the desired amount will be reduced to match the maximum budget constraint.

This works somehow but still, we see  some rounding errors. Here's a screenshot:

issue32

We see a remaining budget of 0.01€. So allocating 7.00€ for broadcast `Forschung aktuell` should be OK. But the frontend reduces the desired amount to *6.99*.

I just pushed a frontend test, that will expose the bug with this very situation.

**Comment by roschaefer** [comment]
This [test](https://github.com/roschaefer/rundfunk-mitbestimmen/blob/master/frontend/tests/unit/models/invoice-test.js#L141) just checks that you can allocate Euros and Cents but no smaller units. It allocates `5.326` and expects a valid distribution probably given by some kind of truncation.

**Comment by domangi** [comment]
Yes, the test passed after I added a 2 decimal truncation to the desired amount e.g. 5.326 becomes 5.32, instead of rounding to 5.3.

File: frontend/app/models/invoice.js [hunk 1]
```diff
@@ -24,7 +24,10 @@ export default DS.Model.extend({
     value = value.toString().split('e');
     return +(value[0] + 'e' + (value[1] ? (+value[1] + exp) : exp));
   },
-
+  twoDecimalDifference(desired, current){
```
Comment (roschaefer) [inline frontend/app/models/invoice.js]: According to the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed#Return_value) `toFixed` returns a string. Can we use the [decimal adjustment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/floor) method maybe?

**Comment by roschaefer** [comment]
After some experimentation, it seems, we must use `round` instead of `floor` for the subtraction. The PR looks nice and ready to merge!

**Comment by domangi** [comment]
close #32

**Pull request merged**